client/app/telecom/telephony/alias/portability/order/telecom-telephony-alias-portability-order.controller.js

```diff
@@ -1,242 +1,255 @@
+          ? _.first(self.typologies[self.order.country]).value : null;
+        self.order.type = self.isSpecialNumber ? 'special' : 'landline';
+
+        self.order.translatedCountry = $translate.instant(`telephony_alias_portability_order_contact_country_${self.order.country}`);
+      };
+
+      self.onChooseRedirectToLine = function (result) {
+        self.order.lineToRedirectAliasTo = result.serviceName;
+        self.order.lineToRedirectAliasToDescription = result.description;
+      };
+
+      // add sdaNumberToAdd number to numbersList
+      self.addSdaNumber = function () {
+        self.order.numbersList.push(self.order.sdaNumberToAdd);
+        self.order.numbersList = _.uniq(self.order.numbersList);
+        self.order.sdaNumberToAdd = null;
+      };
+
+      // remove given number from numbersList
+      self.removeSdaNumber = function (number) {
+        self.order.numbersList = _.pull(self.order.numbersList, number);
+      };
+
+      // normalize number : replace +33 by 0033
+      self.normalizeNumber = function (numberParam) {
+        let number = numberParam;
+        if (number) {
+          number = number.replace(/^\+/, '00');
+        }
+        return number;
+      };
+
+      self.goToConfigStep = function () {
+        self.order.addressTooLong = false;
+
+        if ((_.get(self.order, 'streetName', '').length + _.get(self.order, 'streetNumber', '').length + _.get(self.order, 'streetNumberExtra', '').length + _.get(self.order, 'streetType', '').length) >= 35) {
+          self.order.addressTooLong = true;
+          return false;
+        }
+
+        self.step = 'config';
+        return true;
+      };
+
+      self.getOrderParams = function () {
+        let params = _.pick(self.order, sharedAttributes);
+
+        if (params.offer === 'individual') {
+          params = _.assign(params, _.pick(self.order, 'rio'));
+        } else {
+          params = _.assign(params, _.pick(self.order, 'siret'));
+        }
+        params.firstName = params.firstName || '';
+
+        if (self.isSDA && self.order.sdatype === 'select' && self.order.numbersList.length) {
+          params.listNumbers = _.map(self.order.numbersList, self.normalizeNumber).join(',');
+        }
+
+        if (self.isSpecialNumber) {
+          params.specialNumberCategory = self.order.specialNumberCategory.replace('fr_', '');
+        }
+
+        if (params.country === 'france') {
+          params.fiabilisation = self.order.sdatype === 'all';
+        }
+
+        params.callNumber = self.normalizeNumber(params.callNumber);
+        params.contactNumber = self.normalizeNumber(params.contactNumber);
+        params.desireDate = moment(params.desireDate).format('Y-MM-DD');
+
+        return params;
+      };
+
+      self.fetchPriceAndContracts = function () {
+        self.step = 'summary';
+        return OvhApiOrder.Telephony().v6()
+          .getPortability(self.getOrderParams()).$promise
+          .then((result) => {
```

Comment:
You can destructure `result` :  
```suggestion
          .then(({details, contracts, prices}) => {
            self.details = details;
            self.contracts = contracts;
            self.prices = prices;
```

Reply:
previous code